### PR TITLE
Add support for OAuth login flow using authorization_grant with PKCE

### PR DESCRIPTION
**Pull request by rli**
This change allows the PCKE login flow to be used after enabling a developer flag. `sdk-codegen` is untouched other than bringing in the new service model from internal and the existing SDK customization from public GitHub.

During testing, a classpath conflict on Netty was discovered and it appears that we can safely exclude it from our plugin distribution.

On the token cache side, we enforce distinction during de/serialization since the registration/access token pairs are not compatible with each other (PKCE requires new client registration instances).

Otherwise the entirety of the PKCE grant can be expressed in `ToolkitOAuthService.kt` by overriding the relevant abstract methods in the standard IDE base classes.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt [hunk 1]
```diff
@@ -63,37 +85,79 @@ class SsoAccessTokenProvider(
...
+
+        return registeredClient
+    }
+
+    private fun registerPkceClient(): PKCEClientRegistration {
+        loadPkceClientRegistration()?.let {
+            return it
+        }
+
+        if (!ssoUrl.contains("identitycenter")) {
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt:129]: Should we have a test for this if we are doing text parsing?

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        if (toolkitRequest.csrfToken != callbackState) {
+            request.result.completeExceptionally(RuntimeException("Invalid CSRF token"))
+            return false
+        }
+
+        return super.handleServerCallback(path, parameters)
+    }
+
+    override fun revokeToken(token: String) {
+        TODO("Not yet implemented")
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt:69]: nit: Should this be commented?

**Review by manodnyab — APPROVED** [review]

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt (hunk 1 above)
Comment (rli) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt:129]: it's just a warning, service is right place to make the final call

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt (hunk 2 above)
Comment (rli) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt:69]: could not figure out what calls this

**Pull request merged**